balena/models/device_os.py

```diff
@@ -259,37 +310,145 @@ def get_device_os_semver_with_variant(self, os_version, os_variant=None):
+                        version=version,
+                        non_variant_builds='.'.join(non_variant_build_parts)
+                    )
+
+                variant = next((variant for variant in release['version']['build'] if variant in os_variant_keywords) , None)
+                # Backfill the native rel
+				# TODO: This potentially generates an invalid semver and we should be doing
+				# something like `.join(!version.includes('+') ? '+' : '.')`,  but this needs
+				# discussion since otherwise it will break all ESR released as of writing this.
+                release['raw_version'] = '.'.join([x for x in [version, variant] if x])
+
+            based_on_version = tag_map[self.BASED_ON_VERSION_TAG_NAME] if self.BASED_ON_VERSION_TAG_NAME in tag_map else version
+            line = self.__get_os_version_release_line(version, app_tags)
+
+            release.update({
+                'os_type': app_tags['os_type'],
+                'line': line,
+                'stripped_version': version,
+                'based_on_version': based_on_version,
+                'variant': variant,
+                'formatted_version': 'v{version}{line}'.format(version=version,line=f' ({line})' if line else '')
+            })
+            releases_with_os_versions.append(release)
+
+        return releases_with_os_versions
+                    
+    def __transform_host_apps(self, host_apps):
+        os_versions_by_device_type = {}
+
+        for host_app in host_apps:
+            host_app_device_type = host_app['is_for__device_type'][0]['slug'] if len(host_app['is_for__device_type']) > 0 else None
+
+            if not host_app_device_type:
+                return {}
+            
+            if host_app_device_type not in os_versions_by_device_type:
+                os_versions_by_device_type[host_app_device_type] = []            
+
+            app_tags = self.__get_os_app_tag(host_app['application_tag'])
+            os_versions_by_device_type[host_app_device_type].extend(self.__get_os_versions_from_releases(host_app['owns__release'], app_tags))
+
+        for device_type in os_versions_by_device_type:
+            os_versions_by_device_type[device_type].sort(reverse=True, key=cmp_to_key(sort_version))
+            recommended_per_os_type = {}
+            
+            for version in os_versions_by_device_type[device_type]:
+                if version['os_type'] not in recommended_per_os_type:
+                    if version['variant'] != 'dev' and not version['known_issue_list'] and not version['version']['prerelease']:
+                        additional_format = ' ({line}, recommended)'.format(line=version['line']) if version['line'] else ' (recommended)'
+                        version['is_recommended'] = True
+                        version['formatted_version'] = 'v{version}{additional_format}'.format(version=version['stripped_version'], additional_format=additional_format)
```

Comment:
@thgreasi, I still see these properties in the master branch of JS SDK, will they be removed soon?

Reply:
@nghiant2710 I've already deprecated them in the current version and going to drop them in the next major.
See: https://github.com/balena-io/balena-sdk/blob/cc5a64f364876d6b9d0585c27f2eadc9c936114f/lib/models/os.ts#L73-L76
The reasoning is that if we want to move closer to the model & that the latest version should be the one that's always recommended by our clients anyway.
The CLI & UI was using those props so far but they have now moved away afair, so given that no client depends on them on the python-sdk I was thinking we could completely avoid adding them and then removing them to be in sync with the node sdk (as it will be after the majro bump).